stdlib/public/core/Optional.swift

```diff
@@ -167,7 +196,37 @@ public enum Optional<Wrapped>: ExpressibleByNilLiteral {
       return .none
     }
   }
+}
+
+extension Optional where Wrapped: ~Copyable {
+  @_alwaysEmitIntoClient
+  public consuming func consumingMap<U: ~Copyable>(
+    _ transform: (consuming Wrapped) throws -> U
+  ) rethrows -> U? {
```

Comment:
Do we want these to be typed throws?

Reply:
Ooh, yes we definitely do!

I'll leave a FIXME on `map`. Cc @glessard